### PR TITLE
 feat(ST-65): add get enrolled students api from a classrooms

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added a GET `classrooms/:id/students` API in classrooms controller
- Added a function to get all the students from the classroom with the id

## Associated Ticket Link(s)

- https://trello.com/c/PownKChW

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/classrooms/classrooms.service.ts [hunk 1]
```diff
@@ -86,4 +87,21 @@ export class ClassroomsService {
 
     return this.classroomsStudentsRepository.deleteOne(classroomStudent);
   }
+  async getClassroomStudents(id: number): Promise<IStudentInfo[]> {
+    console.log(id);
+
+    const enrolledStudents = await this.classroomsStudentsRepository.find(
+      { classroomId: id },
+      { populate: ["studentId.user"] },
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:95]:
```suggestion
      { populate: ["student.user"] },
```

Can't we do this?

File: src/classrooms/classrooms.service.ts [hunk 2]
```diff
@@ -86,4 +87,21 @@ export class ClassroomsService {
...
+    );
+
+    const studentsWithUserInfo = enrolledStudents.map((classroomStudent) => ({
+      studentId: classroomStudent.studentId.id,
+      firstName: classroomStudent.studentId.user.firstName,
+      lastName: classroomStudent.studentId.user.lastName,
+      email: classroomStudent.studentId.user.email,
+    }));
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:103]: Is this doing what a serializer should do?

Comment (ishmam-iqbal-sazim) [reply src/classrooms/classrooms.service.ts:103]: Please use serializer at the controller level

File: src/classrooms/classrooms.service.ts (hunk 1 above)
Comment (faysal-alam-sazim) [reply src/classrooms/classrooms.service.ts:95]: No bhai.

File: src/classrooms-students/classrooms-students.repository.ts [hunk 3]
```diff
@@ -5,4 +5,9 @@ import { EntityRepository } from "@mikro-orm/postgresql";
 import { ClassroomStudent } from "@/common/entities/classrooms-students.entity";
 
 @Injectable()
-export class ClassroomsStudentsRepository extends EntityRepository<ClassroomStudent> {}
+export class ClassroomsStudentsRepository extends EntityRepository<ClassroomStudent> {
+  async deleteOne(clssroomStudent: ClassroomStudent) {
+    await this.em.removeAndFlush(clssroomStudent);
+    return { message: "Removed Student from Classroom!" };
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms-students/classrooms-students.repository.ts:11]: Shouldn't be seeing this here. I'm pretty sure I commented this in a PR before. Please handle messages to be displayed from the client

File: src/classrooms/classrooms.controller.ts [hunk 4]
```diff
@@ -45,4 +58,11 @@ export class ClassroomsController {
   ) {
     return this.classroomsService.removeStudentFromClassroom(user.id, deleteEnrollDto);
   }
+
+  @Get(":id/students")
+  async getClassroomStudents(@Param("id") id: string) {
+    const students = await this.classroomsService.getClassroomStudents(Number(id));
+    // return students;
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.controller.ts:65]: Let's remove comments

File: src/users/users.controller.ts [hunk 5]
```diff
@@ -49,4 +51,12 @@ export class UsersController {
   me(@CurrentUser() user: ITokenizedUser) {
     return user;
   }
+
+  @Get("students")
+  async getAllStudents() {
+    const users = await this.usersService.getAllStudents();
+    const studentsInfo = this.usersSerializer.serializeMany(users);
```
Comment (ishmam-iqbal-sazim) [inline src/users/users.controller.ts:58]: Unnecessary var. You can just return it here

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**